executor/linux/build.go

```diff
@@ -491,6 +491,50 @@ func (c *client) ExecBuild(ctx context.Context) error {
 	return c.err
 }
 
+// StreamBuild receives a StreamRequest and then
+// runs StreamService or StreamStep in a goroutine.
+func (c *client) StreamBuild(ctx context.Context) error {
+	// create an error group with the parent context
+	//
+	// https://pkg.go.dev/golang.org/x/sync/errgroup?tab=doc#WithContext
+	streams, streamCtx := errgroup.WithContext(ctx)
+
+	defer func() {
+		c.Logger.Trace("waiting for stream functions to return")
+
+		err := streams.Wait()
+		if err != nil {
+			c.Logger.Errorf("error in a stream request, %v", err)
+		}
+
+		c.Logger.Trace("all stream functions have returned")
```

Comment:
I'm debating. Should this be a different level like Info? That way you can see that each of the stream goroutines have returned in prod.
```suggestion
		c.Logger.Info("all stream functions have returned")
```

Reply:
after running a few test runs, i agree. that is nice confirmation.